src/view/com/composer/Composer.tsx

```diff
@@ -129,6 +130,17 @@ export const ComposePost = observer(function ComposePost({
   const {closeAllDialogs} = useDialogStateControlContext()
   const t = useTheme()
 
+  // Disable this in the composer to prevent any extra keyboard height being applied.
+  // See https://github.com/bluesky-social/social-app/pull/4399
+  const {setEnabled} = useKeyboardContext()
```

Comment:
Just a small hint - for toggling enabled/disabled states it's better to use `useKeyboardController` hook. Was there any reasons why you preferred to consume context directly?

I mean, it's not forbidden and eventually `useKeyboardController` is also a simple context consumer, but consuming context directly it's kind of a last resort and was designed to be used in class-components (and in general `useKeyboardContext` is not mentioned in API section, so it can be a subject to internal changes). For functional components I think it would be better to use `useKeyboardController`  hook which has a defined area of responsibility (toggling enabled/disabled state).

But let me know if I'm missing something important and in which cases better to consume a context 🙌

Reply:
Ah, that's good to know! We ran into some nasty problems yesterday and had to get something through quickly. I had been toying around with the library the other day and knew about the context but wasn't aware of the `useKeyboardController` hook itself 😅. Thanks for catching that! Have not tried it yet but I assume it works the same way according to the docs 🙌

Reply:
Switched this here, thanks for the tip 🙏 https://github.com/bluesky-social/social-app/pull/4458/commits/a5434e9b10bfe5b3f495f018d1294f06b8e7c9be